### ISSUE TITLE
Homework 1 grade

### ISSUE THREAD

**Issue by jarretrt** [aliforgetti/ml-assign#1]
Hi Ali,

There are a couple issues with your submission that need to be corrected. 

1) The predictions for the tilted loss functions should be made through specifying a separate tilted loss function, similar to your functions "L1", "L2" that is then passed on to the "fit_lin" function. Please check the lecture on decision theory (lecture-2.pdf) for an expression of the tilt loss function. Currently, your code is minimizing the quantiles of the loss function, rather than the expected tilted loss function Within the "fit_lin" function, you should still have the form 

mean(loss(y,  beta[1] + beta[2]*x))

rather than 

quantile(loss(y,  beta[1] + beta[2]*x), tau)

When the argument "tau" is specified, it should be passed on to the loss function, rather than quantile(). So it could look something like 

mean(loss(y,  beta[1] + beta[2]*x, tau))

where tau is passed into the "fit_lin" function. Alternately, "tau" could be specified with a value within the loss function itself. 

2) The "loss" argument in the fitting functions should be an actual function, rather than a character string naming the function (e.g. loss = L1 rather than loss = "L1"). This way, you can pass in any pre-specified loss function, rather than needing to anticipate all possibilities in the fitting function. 

I'll be happy to talk further about this during office hours if you have questions. 

Thanks,
Ryan

**Comment by aliforgetti** [comment]
I have fixed the errors that you mentioned and resubmitted. Please let me know if it looks okay! :)

**Comment by jarretrt** [comment]
Hey Ali, 

Your homework does look correct to me. That said, I'm really sorry but we're only allowing one week post-feedback to resubmit corrections, so I'm afraid I can't pass you for the assignment. I apologize for not closing this earlier. 

Best,
Ryan
